### PR TITLE
 [WFLY-13793]: Allow remote jms queue/topic not to use legacy amq1 prefix.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding new mode attribute for external destinations.
 * Adding new supported property for @JMSDestination
 * Adding tests
 * Adding documentation.

Jira: https://issues.redhat.com/browse/WFLY-13793

### CODE REVIEW

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/QueueReadAttributeHandler.java [hunk 1]
```diff
@@ -111,6 +112,8 @@ public void executeRuntimeStep(OperationContext context, ModelNode operation) th
             context.getResult().set(control.getMessageCount());
         } else if (SCHEDULED_COUNT.getName().equals(attributeName)) {
             context.getResult().set(control.getScheduledCount());
+        } else if (ROUTING_TYPE.getName().equals(attributeName)) {
```
Comment (soul2zimate) [inline messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/QueueReadAttributeHandler.java:115]: This overlaps fix in subsequent https://github.com/wildfly/wildfly/pull/14417

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/deployment/JMSDestinationDefinitionInjectionSource.java [hunk 2]
```diff
@@ -189,7 +190,7 @@ private void startQueue(final String queueName,
         final boolean durable = properties.containsKey(DURABLE.getName()) ? Boolean.valueOf(properties.get(DURABLE.getName())) : DURABLE.getDefaultValue().asBoolean();
         final String managementAddress =  properties.containsKey(MANAGEMENT_ADDRESS.getName()) ? properties.get(MANAGEMENT_ADDRESS.getName()) : MANAGEMENT_ADDRESS.getDefaultValue().asString();
         final String user =  properties.containsKey("management-user") ? properties.get("management-user") : null;
-        final String password =  properties.containsKey("management-password") ? properties.get("\"management-password") : null;
+        final String password =  properties.containsKey("management-password") ? properties.get("management-password") : null;
```
Comment (soul2zimate) [inline messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/deployment/JMSDestinationDefinitionInjectionSource.java:193]: This overlaps fix made in subsequent https://github.com/wildfly/wildfly/pull/14416

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/jms/external/prefix/ExternalJMSDestinationDefinitionMessagingDeploymentTestCase.java [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+            } else {
+                op.get("connectors").add("http-test-connector");
+            }
+            op.get("enable-amq1-prefix").set(false);
+            execute(managementClient, op, true);
+            op = Operations.createRemoveOperation(getClientTopicAddress());
+            managementClient.getControllerClient().execute(op);
+            op = Operations.createRemoveOperation(getClientQueueAddress());
+            managementClient.getControllerClient().execute(op);
+            ops.close();
```
Comment (soul2zimate) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/jms/external/prefix/ExternalJMSDestinationDefinitionMessagingDeploymentTestCase.java:121]: Neither `ActiveMQProviderJMSOperations.java`  nor `RemoteActiveMQProviderJMSOperations.java` implements this `close()` method 
```
    @Override
    public void close() {
        // no-op
        // DO NOT close the management client. Whoever passed it into the constructor should close it
    }
```

but I think it won't do any harm.

**Review by soul2zimate — APPROVED** [review]
@ehsavoie Thank you for the detailed tests. Overall this LGTM, except a few minor comments.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/jms/external/prefix/ExternalJMSDestinationDefinitionMessagingDeploymentTestCase.java (hunk 3 above)
Comment (ehsavoie) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/jms/external/prefix/ExternalJMSDestinationDefinitionMessagingDeploymentTestCase.java:121]: yes i agree

File: messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalJMSQueueDefinition.java [hunk 4]
```diff
@@ -46,15 +48,18 @@ public ExternalJMSQueueDefinition(boolean registerRuntimeOnly) {
...
-        return Collections.singletonList(CommonAttributes.DESTINATION_ENTRIES);
+        return Arrays.asList(CommonAttributes.DESTINATION_ENTRIES, External.ENABLE_AMQ1_PREFIX);
     }
 
     @Override
     public void registerAttributes(ManagementResourceRegistration registry) {
         if (registerRuntimeOnly) {
             registry.registerReadOnlyAttribute(CommonAttributes.DESTINATION_ENTRIES, null);
+            // Should this be read only as entries ?
+            registry.registerReadOnlyAttribute(External.ENABLE_AMQ1_PREFIX, null);
```
Comment (soul2zimate) [inline messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalJMSQueueDefinition.java:59]: I wasn't really sure on this "readonly" in the past, is that correct to you @ehsavoie  ?

Comment (ehsavoie) [reply messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalJMSQueueDefinition.java:59]: Yes , this is the runtime description of the resource

**Pull request merged**